### PR TITLE
[json-config-next] Reference Imperative ./lib folder in test CLIs

### PR DESCRIPTION
**Pull request by t1m0thyj**
Use "references" property in tsconfig.json (available since TS 3.0) to share the Imperative build (in "./lib") for test CLIs to use.

With this PR, it's no longer necessary to rebuild the test CLIs every time a change is made in Imperative that affects them. The CLIs load Imperative from "./lib" and no longer build their own copies of Imperative.

This also fixes "missing package.json" errors that happen when a test CLI is run manually outside of an integration test, since the "lib" dir of the test CLIs matches the structure that Imperative expects.

The behavior of the "references" property is documented here: https://www.typescriptlang.org/docs/handbook/project-references.html

### CODE REVIEW

**Comment by zFernand0** [comment]
`Do not merge this until #478 has been merged`
Should the base of the PR be json-config-next (since #478 is `next-uc2 -> next`)?

File: __tests__/__integration__/cmd/src/cli/auto-format/object/Object.definition.ts
Comment (zFernand0) [inline __tests__/__integration__/cmd/src/cli/auto-format/object/Object.definition.ts:12]: curious as to why we specify `/index` in the path only for some files and not all.

File: __tests__/__integration__/imperative/tsconfig.json
Comment (zFernand0) [inline __tests__/__integration__/imperative/tsconfig.json:27]: 👍

File: __tests__/__integration__/hello_world/tsconfig.json
Comment (zFernand0) [inline __tests__/__integration__/hello_world/tsconfig.json:36]: Probably a dumb question, ...
doesn't `hello_world` need the `references` as well?

**Review by zFernand0 — APPROVED** [review]
LGTM

File: __tests__/__integration__/cmd/src/cli/auto-format/object/Object.definition.ts
Comment (t1m0thyj) [reply __tests__/__integration__/cmd/src/cli/auto-format/object/Object.definition.ts:12]: Not sure 🙂 For this PR I wanted to keep the amount of changes minimal. Perhaps in the future we should evaluate all of our imports?

File: __tests__/__integration__/hello_world/tsconfig.json
Comment (t1m0thyj) [reply __tests__/__integration__/hello_world/tsconfig.json:36]: Not a dumb question, I was dumb here 😁

**Comment by t1m0thyj** [comment]
> `Do not merge this until #478 has been merged`
> Should the base of the PR be json-config-next (since #478 is `next-uc2 -> next`)?

Eventually yes, for now the base is json-config-next-uc2 to make this PR easier to review 🙂

**Comment by zFernand0** [comment]
PR still LGTM! :+1:

**Review by awharn — APPROVED** [review]
Looks good to me! 👍🏼

**Pull request merged**
